lib/iex/lib/iex/info.ex

```diff
@@ -290,6 +290,32 @@ defimpl IEx.Info, for: Reference do
   end
 end
 
+defimpl IEx.Info, for: [Date, Time, NaiveDateTime] do
+  sigils = [{Date, "D"}, {Time, "T"}, {NaiveDateTime, "N"}]
+  @sigil sigils[@for]
+  reprs = [{Date, "date"}, {Time, "time"},
+           {NaiveDateTime, ~S{"naive" datetime - i.e. a datetime without a timezone}}]
+  @repr reprs[@for]
+
+  def info(value) do
+    desc = """
+    This is a struct representing a #{@repr}. This structs is commonly represented
+    using the `~#{@sigil}` sigil syntax, that is defined in the `Kernel.sigil_#{@sigil}/2` macro.
+    """
+    ["Data type": inspect(@for),
+     "Description": desc,
+     "Raw representation": raw_inspect(value),
```

Comment:
This gives raw map `%{__struct__: Foo, ...}` and I don't think we need that here. A plain struct works better, imho.